src/main/webapp/dev/js/main/instructorHelp.js

```diff
@@ -25,7 +34,319 @@ function bindPanelCollapseLinksAndAnchor() {
+                    // stop if no further matches are found
+                    if (matchPos === -1) {
+                        break;
+                    }
+                    const match = textToSearch.data.substr(matchPos, keyword.length);
+                    // remaining text corpus after current match that needs to be searched further
+                    const nextTextElem = textToSearch.splitText(matchPos);
+
+                    nextTextElem.data = nextTextElem.data.substr(match.length);
+                    highlightMatch(match, nextTextElem);
+                    textToSearch = nextTextElem;
+                }
+            }
+
+            // iterate through each text node and highlight matches
+            childTextElements.each(function () {
+                searchAndHighlightMatches(this);
+            });
+        }
+
+        $('#searchResults').find('.panel.panel-default').each(function () {
+            // highlight heading and body matches for query
+            highlighter(this);
+        });
+    }
+
+    function initializePagingControls() {
+        const numResults = results.length;
+        numPages = Math.ceil(numResults / numResultsPerPage);
+
+        if (numPages < 2) {
+            return;
+        }
+
+        let buttonHtml = '<button type="button" class="btn btn-default" id="prevPage" onclick="renderPage(\'prev\')">'
+                         + '<span class="glyphicon glyphicon-chevron-left"></span></button>';
+        for (let i = 0; i < numPages; i += 1) {
+            buttonHtml += `<button type='button' class='btn btn-default' onclick='renderPage(${i + 1})'>${i + 1}</button>`;
+        }
+        buttonHtml += '<button type="button" class="btn btn-default" id="nextPage" onclick="renderPage(\'next\')">'
+                     + '<span class="glyphicon glyphicon-chevron-right"></span></button>';
+        $('#pagingControls').html(buttonHtml);
+        $('#pagingControls').addClass('padding-15px margin-bottom-35px');
+    }
+
+    // highlight each keyword of query in search results
+    $.each(query.split(' '), (idx, keyword) => {
+        if (keyword.length > 2) {
+            highlightKeyword(keyword);
+        }
+    });
+
+    initializePagingControls();
+    // render first page by default
+    renderPage(1);
+    $('#pagingDivider').show();
+    $('#searchResults').show();
+    $('#searchMetaData').text(`${results.length} results found for '${query}'`);
+    $('#allQuestions').hide();
+    $('#topics').hide();
+}
+
+/**
+ * Simulates clicking of search button on pressing Enter key.
+ */
+function bindEnterKeyForSearchBox() {
+    $('#searchQuery').keypress((e) => {
+        if (e.keyCode === 13) {
+            $('#search').click();
+        }
+    });
+}
+
 $(document).ready(() => {
     syncPanelCollapseWithUrlHash();
     bindPanelCollapseLinksAndAnchor();
+    prepareQuestionsForSearch();
+    bindEnterKeyForSearchBox();
+    resetSearch();
```

Comment:
Rename this function? It doesn't quite make sense to reset search if the search hasn't happened yet. Perhaps something like `displayDefaultView` (maybe not this exactly) might be more appropriate.

Reply:
Sure. How about `displayDefaultAndClearSearchBox` or is that too long ?

Reply:
That's good